### ISSUE TITLE
4.0.x-dev: Option to ignore if no files were checked instead of exiting with an error code

### ISSUE THREAD

**Issue by AndreasA** [squizlabs/PHP_CodeSniffer#3556]
**Describe the bug**
In our project we sometimes prepare pipelines to be used for future files.
However, as those files might not yet exist, phpcs could stop with `No files were checked` error. The solution is to disable phpcs temporarily but then once the files are there, it might be forgotten to enable them again.

I think instead of always failing, there should be an option to skip this error.

**To reproduce**
1. Run phpcs against an empty directory.

It fails which in most cases is fine but there should be an option to not fail instead.

**Expected behavior**

Option to ignore if no files were checked.

**Versions (please complete the following information):**
 - OS: MacOS 12.2.1
 - PHP: 8.1
 - PHPCS: 4.0.x-dev
 - Standard: PSR2

**Comment by schlndh** [comment]
I agree. The error is undesirable in scenarios where an external tool passes specific files to PHPCS without knowing whether they are excluded or not (e.g. via `phpcs.xml`). I have some files excluded because some sniffs don't work correctly with them (e.g. mixed HTML+PHP, I'm mainly thinking about external sniffs), or because they're auto-generated (e.g. from SOAP).

Consider these scenarios:

- CI: since PHPCS rules (AFAIK) treat each file independently (unlike e.g. PHPStan) it makes sense to run PHPCS only on changed files instead of all files. But it can happen that only excluded files are changed leading to the error.
- GIT hooks: same thing.
- Phpstorm PHPCS inspection: it runs only on single file. If the file is excluded it shows an error notification with PHPCS 4.

**Comment by rodrigoprimo** [comment]
A related issue has been filed in the new repository: PHPCSStandards/PHP_CodeSniffer#1331. Linking for visibility.
